src/tools/tools/extension.fs

```diff
@@ -356,4 +356,19 @@ type StringExtension =
     [<Extension>]
     static member DeserializeFromJson( s: String ): 'T = 
         use ms = new MemoryStream(System.Text.Encoding.UTF8.GetBytes(s))
-        ms.ReadJson()
\ No newline at end of file
+        ms.ReadJson()
+    /// Helper function to pack a remote exception 
+    /// See: http://blogs.msdn.com/b/brada/archive/2005/03/27/402801.aspx
+    [<Extension>]
+    static member WriteException( x: Stream, ex: Exception ) = 
+        let fmt = System.Runtime.Serialization.Formatters.Binary.BinaryFormatter()
```

Comment:
For exception serialization (catch remote exception thrown), let's use BinaryFormatter(). The reason is that if the formatter Bruno wrote causes any issue, its exception can be captured and sent back in the Exception framework.